streams/src/main/java/org/apache/kafka/streams/state/WindowBytesStoreSupplier.java

```diff
@@ -33,11 +33,22 @@
      * It is also used to reduce the amount of data that is scanned when caching is enabled.
      *
      * @return number of segments
+     * @deprecated since 2.1. Use {@link WindowBytesStoreSupplier#segmentInterval()} instead.
      */
+    @Deprecated
     int segments();
 
     /**
-     * The size of the windows any store created from this supplier is creating.
+     * The size of the segments (in milliseconds) the store has.
+     * If your store is segmented then this should be the size of segments in the underlying store.
+     * It is also used to reduce the amount of data that is scanned when caching is enabled.
+     *
+     * @return size of the segments (in milliseconds)
+     */
+    long segmentInterval();
```

Comment:
This is a meta comment: currently we are leaking some segmented bytes store specific information into the general interface, such as `segmentInterval`, `retainDuplicates`. Suppose a user customized window bytes store, it does not necessarily need to be segmented at all. In addition, the `windowSize` and `retentionPeriod` are used today only because library's specific caching / logging layer relies on it, which is not generally applicable.

Ideally all of these parameters should not need to be instantiated by users, we could consider refactoring them a bit for better custom store support.

Reply:
7106 should be good enough. This is just a self-reminder than an actual comment :)